### PR TITLE
Fix S3 region when not value set in instance

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

Please outline the changes and how this PR fixes the issue.

### Why are the changes needed?

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

**Comment by yuzhu** [comment]
i think if we initialize the s3 client with `AmazonEC2ClientBuilder.standard()` then we won't have this issue. https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html

**Comment by LuQQiu** [comment]
@yuzhu I tested on my side that this fix is sufficient, the standard may also works, can you help test on your side as well?

**Comment by LuQQiu** [comment]
I didn't see standard() modify the s3 region information, looks like it's similar to what we do now
```
  public static AmazonS3ClientBuilder standard() {
    return (AmazonS3ClientBuilder)(new AmazonS3ClientBuilder()).withCredentials(new S3CredentialsProviderChain());
  }
```
```
    AWSCredentialsProvider credentials = createAwsCredentialsProvider(conf);
  public static AWSCredentialsProvider createAwsCredentialsProvider(
      UnderFileSystemConfiguration conf) {
    // Set the aws credential system properties based on Alluxio properties, if they are set;
    // otherwise, use the default credential provider.
    if (conf.isSet(PropertyKey.S3A_ACCESS_KEY)
        && conf.isSet(PropertyKey.S3A_SECRET_KEY)) {
      return new AWSStaticCredentialsProvider(new BasicAWSCredentials(
          conf.get(PropertyKey.S3A_ACCESS_KEY), conf.get(PropertyKey.S3A_SECRET_KEY)));
    }
    // Checks, in order, env variables, system properties, profile file, and instance profile.
    return new DefaultAWSCredentialsProviderChain();
  }
    AmazonS3ClientBuilder clientBuilder = AmazonS3Client.builder()
        .withCredentials(credentials)
```

**Comment by LuQQiu** [comment]
Tested with s3 access key and secret key in ~/.aws/credentials
s3 bucket in us-east-1, us-west-1, Singapore
bin/alluxio runTests
all passed

File: underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java [hunk 1]
```diff
@@ -86,6 +86,9 @@
 public class S3AUnderFileSystem extends ObjectUnderFileSystem {
   private static final Logger LOG = LoggerFactory.getLogger(S3AUnderFileSystem.class);
 
+  /** The special S3 region which can be used to talk to any bucket. */
+  private static final String S3_DEFAULT_REGION = "us-east-1";
```
Comment (yuzhu) [inline underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java:90]: can we use their enum for this? aws sdk has defined enums

**Review by yuzhu — APPROVED** [review]
Thanks @LuQQiu LGTM

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.7 please

**Pull request merged**